src/service/resources-impl.js

```diff
@@ -1493,6 +1518,8 @@ export class Resources {
       });
       this.requestsChangeSize_ = this.requestsChangeSize_.filter(
           request => request.resource != resource);
+      this.pendingBuildResources_ = this.pendingBuildResources_.filter(
```

Comment:
@dvoytenko now that I actually think of it, we really shouldn't clean this up when the element unload! This is for building not for layouting so we should always keep it in order for the element to be built otherwise if we clean this and the user come back to the page the element will never be built if it was pending!

Let me know if I am missing something.